File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> mp = new LinkedHashMap<>();
        char[] sArr = s.toCharArray();
        for(char ch: s.toCharArray()) {
            mp.put(ch, mp.getOrDefault(ch,0)+1);
        }
        
        for(int i=0;i<sArr.length;i++) {
            if(mp.get(sArr[i]) == 1)
                return i;
        }
        return -1;
    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let mp = new Map();
    for(let ch of s) 
        mp.set(ch, (mp.get(ch) || 0)+1)
    for(let i=0;i<s.length;i++) 
        if(mp.get(s[i]) == 1)
            return i;
    return -1;
};